WordPress/gutenberg
# Add featured image to Media & Text block

## What?

Add an option to use the featured image in a Media & Text block.
Closes https://github.com/WordPress/gutenberg/issues/35221

## Why?

The Media & Text block has a unique feature called "stack on mobile" which is useful when you want to display the post title and featured image together but with different layouts on desktop and mobile.
This was a feature requested for the "Blue Note" community theme.

## How?

- Adds a new block attribute called "useFeaturedImage". Adds the post id and post type as a new context.
- Deprecates the version that does not have a featured image option
- Adds the featured image option in edit.js and updates the conditions for showing the block options.
- Adds an index.php for outputting the block on the front.
- Updates the media-container utility file to include the featured image option.

The featured image can be selected in the initial block placeholder and from the block toolbar.
A placeholder image is used if the featured image option is chosen, but the post has no featured image.

## Testing Instructions

1. Open the block editor and add a media and text block

Test the following:
Replacing an existing image or video with a featured image
Adding a featured image when there is no featured image on the post or page
Adding a featured image when there is a featured image on the post or page

Removing the featured image from the post/page sidebar
Replacing the featured image from the post/page sidebar
Replacing the featured image from the block toolbar: by uploading a new media and by selecting from the media library

Next, please confirm that:

- Alt text works
- The cropping option and the focal point picker works
- Styling and aligning the block works
- Linking the image works

### Testing Instructions for Keyboard


## Screenshots or screencast 

In this video, I start with a post that has a featured image. I add a new media & text block, enable the featured image option, and try some block toolbar settings. Then I replace the featured image with an image from the media library and then replace that image with the featured image again.

https://github.com/WordPress/gutenberg/assets/7422055/bb0e6d92-1776-4e05-bf3d-5263303e62ac

In this video, I am enabling the featured image option even though the post has no featured image, showing the placeholder:

https://github.com/WordPress/gutenberg/assets/7422055/99970fbb-c4df-4de1-a9b9-e167096bc19c

In this video, I have added the featured image and an anchor to the block, and I am checking that the anchor is added on the front.

https://github.com/WordPress/gutenberg/assets/7422055/f1c15063-7c0d-45a5-bbc2-a313ffadcbe0

Twenty Twenty Three blog / Home template with featured image and post title in the media & text block inside the query:


https://github.com/WordPress/gutenberg/assets/7422055/f27aa804-04a7-4f2a-835b-18f40be31952
